### PR TITLE
Bg/tensor attribute value

### PR DESCRIPTION
**Pull request by bghimireamd**
## Proposed changes

Update frontend schema & API to handle Tensor_attributes as values

## Checklist

Please put an `x` into the boxes that apply. You can also fill these out after creating the PR. If you're not sure, please don't hesitate to ask.

- [x] I have added automated tests relevant to the introduced functionality
- [x] I have sufficient test coverage for the changes, and code coverage hasn't decreased as a result of my PR
- [x] I have ran the tests, and they are all passing locally
- [ ] I have added relevant documentation for the changes
- [ ] I have removed the stale documentation which is no longer relevant after this pull request
- [x] I have ran `make format` & `make check_format` to ensure the changes have been formatted

### CODE REVIEW

File: frontend/include/hipdnn_frontend/attributes/tensor_attributes.hpp [hunk 1]
```diff
@@ -15,6 +20,41 @@ namespace graph
 class Tensor_attributes
 {
 public:
+    using ValueVariant = std::variant<std::monostate, float, uint16_t, uint8_t, int32_t>;
```
Comment (mousdahl-amd) [inline frontend/include/hipdnn_frontend/attributes/tensor_attributes.hpp]: Do we want to add a follow-up ticket to switch these to half / bfloat16 after I merge?

Comment (mousdahl-amd) [reply frontend/include/hipdnn_frontend/attributes/tensor_attributes.hpp]: No double?

Comment (BrianHarrisonAMD) [reply frontend/include/hipdnn_frontend/attributes/tensor_attributes.hpp]: Yea we should, and likely add double.

File: sdk/schemas/tensor_attributes.fbs [hunk 2]
```diff
@@ -5,11 +5,35 @@ include "data_types.fbs";
...
+table IValue {
+  value: int;
+}
+
+union Tensor_Value {
+  FValue,
+  HValue,
+  UValue,
+  IValue
+}
```
Comment (BrianHarrisonAMD) [inline sdk/schemas/tensor_attributes.fbs]: Can't you have a union of primitive types?

File: frontend/include/hipdnn_frontend/attributes/tensor_attributes.hpp [hunk 3]
```diff
@@ -129,8 +169,50 @@ class Tensor_attributes
     flatbuffers::Offset<hipdnn_sdk::data_objects::TensorAttributes>
         pack_attributes(flatbuffers::FlatBufferBuilder& builder) const
     {
-        return CreateTensorAttributesDirect(
-            builder, _uid, _name.c_str(), to_sdk_type(_data_type), &_stride, &_dim, _is_virtual);
+        // using FB = hipdnn_sdk::data_objects;
+
```
Comment (BrianHarrisonAMD) [inline frontend/include/hipdnn_frontend/attributes/tensor_attributes.hpp]: Can remove this I believe.

File: frontend/tests/test_tensor_value_attributes.cpp
Comment (BrianHarrisonAMD) [inline frontend/tests/test_tensor_value_attributes.cpp:1]: We should have some tests verifying it fails to retrieve values if you pass in one type, and request another back out. 

For example, if you set a float, and request a uint8 it should fail.

File: sdk/schemas/tensor_attributes.fbs [hunk 4]
```diff
@@ -5,11 +5,35 @@ include "data_types.fbs";
...
+
 table TensorAttributes {
     uid: long;
     name: string;
     data_type: DataType;
     strides: [long];
     dims: [long];
     virtual: bool;
+    value: Tensor_Value;
```
Comment (BrianHarrisonAMD) [inline sdk/schemas/tensor_attributes.fbs:38]: Not sure why I did this, but the naming of the other tables seems to be different than this.
We should unify the naming.

File: sdk/schemas/tensor_attributes.fbs [hunk 5]
```diff
@@ -5,11 +5,35 @@ include "data_types.fbs";
...
+
+table UValue {
+  value: ubyte;
+}
+
+table IValue {
+  value: int;
+}
+
+union Tensor_Value {
```
Comment (BrianHarrisonAMD) [inline sdk/schemas/tensor_attributes.fbs]: Not sure why I did this, but the naming of the other tables seems to be different than these.
We should make the naming for these consistent.

File: frontend/tests/test_tensor_value_attributes.cpp
Comment (BrianHarrisonAMD) [reply frontend/tests/test_tensor_value_attributes.cpp:1]: Also a test for packing, and unpacking an empty value I guess.

**Review by BrianHarrisonAMD — APPROVED** [review]
LGTM

File: sdk/schemas/tensor_attributes.fbs (hunk 2 above)
Comment (BrianHarrisonAMD) [reply sdk/schemas/tensor_attributes.fbs]: I believe you cannot.
